pkg/resmgr/resource-manager.go

```diff
@@ -271,10 +271,12 @@ func (m *resmgr) startControllers() error {
 
 // updateTopologyZones updates the 'topology zone' CRDs.
 func (m *resmgr) updateTopologyZones() {
-	m.Info("updating topology zones...")
-	err := m.agent.UpdateNrtCR(m.policy.ActivePolicy(), m.policy.GetTopologyZones())
-	if err != nil {
-		m.Error("failed to update topology zones: %v", err)
+	if len(m.policy.GetTopologyZones()) != 0 {
+		m.Info("updating topology zones...")
+		err := m.agent.UpdateNrtCR(m.policy.ActivePolicy(), m.policy.GetTopologyZones())
+		if err != nil {
+			m.Error("failed to update topology zones: %v", err)
+		}
```

Comment:
nit: I would update this to query topology zone once, check if we got data and if we do, pass the already queried data to the agent for publishing. IOW, something like
```
  if zones := m.policy.GetTopologyZones(); len(zones) != 0 {
    m.Info("updating topology zones...")
    if err := m.agent.UpdateNrtCR(m.policy.ActivePolicy(), zones);  err != nil {
      m.Error("failed to update topology zones: %v" ,err)
    }
  }
```

Reply:
fixed.